js/ui/grid_core/ui.grid_core.editing.js

```diff
@@ -1189,9 +1267,19 @@ const EditingController = modules.ViewController.inherit((function() {
                 data: item && item.data,
                 cancel: false
             };
-            const oldEditRowIndex = that._getVisibleEditRowIndex();
             const visibleColumns = columnsController.getVisibleColumns();
-            const oldColumn = visibleColumns[that._editColumnIndex];
+            const isEditByOptionChanged = isDefined(oldColumnIndex) || isDefined(oldRowIndex);
+
+            let oldColumn;
+            if(isDefined(oldColumnIndex)) {
+                oldColumn = visibleColumns[oldColumnIndex];
+            } else {
+                oldColumn = this._getEditColumn();
+            }
+
+            if(!isDefined(oldRowIndex)) {
+                oldRowIndex = that._getVisibleEditRowIndex();
+            }
```

Comment:
The _editCellCore method became big. Extract options processing to separate method like _normalizeEditCellOptions.